src/client.ts

```diff
@@ -0,0 +1,355 @@
+        const bigIntPropertyKeys: (keyof BlockInfo)[] = [
+            'blockHeight',
+            'blockBaker',
+            'blockSlot',
+            'transactionEnergyCost',
+            'transactionCount',
+            'transactionsSize',
+        ];
+
+        return unwrapJsonResponse<BlockInfo>(
+            response,
+            buildJsonResponseReviver(datePropertyKeys, bigIntPropertyKeys),
+            intToStringTransformer(bigIntPropertyKeys)
+        );
+    }
+
+    /**
+     * Retrieves the blocks are the given height.
+     * @param height the block height as a positive integer
+     * @returns a string array containing the blocks at the given height, i.e. ['blockHash1', 'blockHash2', ...]
+     */
+    async getBlocksAtHeight(height: bigint): Promise<string[]> {
+        if (height <= 0n) {
+            throw new Error(
+                'The block height has to be a positive integer, but it was: ' +
+                    height
+            );
+        }
+        const blockHeight = new BlockHeight();
+        blockHeight.setBlockHeight(height.toString());
+        const response = await this.sendRequest(
+            this.client.getBlocksAtHeight,
+            blockHeight
+        );
+        return unwrapJsonResponse<string[]>(response);
+    }
+
+    /**
+     * Retrieves the consensus status information from the node. Note that the optional
+     * fields will only be unavailable for a newly started node that has not processed
+     * enough data yet.
+     */
+    async getConsensusStatus(): Promise<ConsensusStatus> {
+        const response = await this.sendRequest(
+            this.client.getConsensusStatus,
+            new Empty()
+        );
+
+        const datePropertyKeys: (keyof ConsensusStatus)[] = [
+            'blockLastReceivedTime',
+            'blockLastArrivedTime',
+            'genesisTime',
+            'lastFinalizedTime',
+        ];
+        const bigIntPropertyKeys: (keyof ConsensusStatus)[] = [
+            'epochDuration',
+            'slotDuration',
+            'bestBlockHeight',
+            'lastFinalizedBlockHeight',
+            'finalizationCount',
+            'blocksVerifiedCount',
+            'blocksReceivedCount',
+        ];
+
+        return unwrapJsonResponse<ConsensusStatus>(
+            response,
+            buildJsonResponseReviver(datePropertyKeys, bigIntPropertyKeys),
+            intToStringTransformer(bigIntPropertyKeys)
+        );
+    }
+
+    sendRequest<T, Response extends Serializable>(
+        command: Command<T, Response>,
+        input: T
+    ): Promise<Uint8Array> {
+        const deadline = new Date(Date.now() + this.timeout);
+        return new Promise<Uint8Array>((resolve, reject) => {
+            if (this.client.waitForReady === undefined) {
+                reject(
+                    new Error('The client is missing the waitForReady function')
+                );
+            } else {
+                this.client.waitForReady(deadline, (error) => {
```

Comment:
I don't understand who defines this waitForReady, Maybe you could explain this to me at some point

Reply:
I will have to double check, it is something I took from the desktop wallet. If I remember correctly it is just because it is not typed correctly, so it actually has that function available, but the type checking fails on it due to the types.

Reply:
I think this might be fixed by making some changes to the proto file generation. I will try a few things.

Reply:
This was fixed by making the proto generator use `@grpc/grpc-js' instead of 'grpc'. It caused some type problems with the `Command<T, Response>` type that I was unable to resolve for now. The typing will have to be fixed later.